src/swarm-engine/Swarm/Game/Step/Path/Finding.hs

```diff
@@ -145,7 +147,8 @@ pathCommand parms = do
   goalReachedFunc loc = case target of
     LocationTarget gLoc -> return $ loc == gLoc
     EntityTarget eName -> do
-      me <- entityAt $ Cosmic currentSubworld loc
+      worlds <- Fused.use $ landscape . multiWorld
+      let me = lookupCosmicEntity (Cosmic currentSubworld $ locToCoords loc) worlds
```

Comment:
I gave up and used pure functions for path finding, which seems to have fixed the issue.